src/main/java/com/limechain/network/StrictProtocolBinding.java

```diff
@@ -11,7 +11,7 @@ public StrictProtocolBinding(String protocolId, P2PChannelHandler<T> protocol) {
         super(protocolId, protocol);
     }
 
-    public T dialPeer(Host us, PeerId peer, AddressBook addrs) {
+    public synchronized T dialPeer(Host us, PeerId peer, AddressBook addrs) {
```

Comment:
Are we sure we want this to be synchronized? How long is the timeout of `dial`? If we need the synchronization what's the drawback of having a synchronized block just around `dial` instead of the whole function being synchonized?
Also always add javadoc to synchronized functions as they are very prone to breaking/causing deadlocks.

Reply:
Not needed. Removed.